### ISSUE TITLE
check address balance - rest request error

### ISSUE THREAD

**Issue by wdy43di** [input-output-hk/jormungandr#869]
**Mandatory Information**
1. `jcli --full-version` output; jcli 0.5.4 (HEAD-a9ec5ce, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)]
2. `jormungandr --full-version` output; jormungandr 0.5.4 (HEAD-a9ec5ce, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)]

**To Reproduce**
root@ashenjormungandr:/jor# y
y: command not found
root@ashenjormungandr:/jor# ./jcli rest v0 account get $(cat receiver_account.txt) -h http://127.0.0.1:3101/api
failed to make a REST request
|-> could not connect with node
|-> http://127.0.0.1:3101/api/v0/account/9f30b10326d3d06debb04d6b01d9d5e649ff3aeaa71667bb9b6b66f1812d8b4d: error trying to connect: Connection refused (os error 111)
|-> Connection refused (os error 111)

root@ashenjormungandr:/jor# uname -srm
Linux 4.15.0-64-generic x86_64
root@ashenjormungandr:/jor# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description: Ubuntu 18.04.3 LTS
Release: 18.04
Codename: bionic

the command ./jcli rest v0 utxo get --host "http://127.0.0.1:3101/api" seems to work.

**Expected behavior**
Following https://github.com/input-output-hk/shelley-testnet/wiki/How-to-setup-a-Jormungandr-Networking--node-%28v-0.5.0%29 trying to find my balance.

Thank you for your time.

**Comment by gufmar** [comment]
if the account address never received any input it is the common behaviour. But Then you should get a "404 Not Found" and not a "connection refused" (what I effectively get when I ask my REST endpoint for this address) ?

However, I want to use this existing issue to report a transaction where it seems the value does not arrive at the destination

To   ca1shm3e833hzsvevmh6ej8ddnardu7mkqx330pzumknhxq25atxla9qvhulqs
Hash 85054ef30b8068ded6e3e5f8018a0cce7f04b8ac45c7283cfeb4baf0793e18e7

In this case also restarting the node didn't help.

**Comment by dkijania** [comment]
are you sure your local node was up at the time you were sending request ?
you've send
```
 ./jcli rest v0 utxo get --host "http://127.0.0.1:3101/api" 
```
After `account get` or before ?

Usually good way to check if node is up is to use command:
```
 ./jcli rest v0 node stats get --host "http://127.0.0.1:3101/api" 
```

**Comment by CodeSandwich** [comment]
@wdy43di I can't reproduce the issue. Dkijania's suggestion that the node may be down seems most plausible, that's what this error means. Could you provide the exact command that causes this issue and check if it still fails? I have no idea what `$(cat receiver_account.txt)` returns on your machine and it could be the key.

**Comment by dkijania** [comment]
@wdy43di
Actually i was able to see similar situation. When i started node in a self-node mode (not connected to any network) everything works ok. However, if I try to bootstrap node with trusted peers from IOHK servers there is a significant delay in node startup. On my machine it's about 4 minutes until rest api responds.

**Comment by CodeSandwich** [comment]
That's right, bootstrapping is long and silent process which happens **before** startup of any node internals. There's no REST endpoint during that phase. Could you check that @wdy43di?

**Comment by CodeSandwich** [comment]
@wdy43di I'm going to assume that the problem was lack of REST in bootstrapping phase. It's already fixed, so I'm going to close this issue.

**Comment by wdy43di** [comment]
thx
